Recruit-Training/Advanced-Recruit-Training/src/BatteryStateOfChargeService.cpp

```diff
@@ -16,23 +22,72 @@ BatteryStateOfChargeService::~BatteryStateOfChargeService()
 
 double BatteryStateOfChargeService::totalAmpHoursUsed() const
 {
-    return 0.0;
+
+    return totalAmphoursUsed_;
 }
 
 bool BatteryStateOfChargeService::isCharging() const
 {
+    if(current_ < 0){
+        return true;
+    }
     return false;
 }
 
 QTime BatteryStateOfChargeService::timeWhenChargedOrDepleted() const
 {
-    return QTime::currentTime();
+    double totalHoursRemaining, msRemaining;
+        int h, m, s, ms;
```

Comment:
Use more descriptive variable names to keep code clean

Reply:
Also, we should separate variable declarations to one per line.